### PR TITLE
Update README for pending release

### LINKED ISSUE #498: Update README.md for v0.4.0
* Has over 240 test cases right now -> Currently has over 420 tests.
 * Update example to use `write_problem()` instead of `write_to_file()`
 * "Can write out the MCNP problem verbatim, and try to match the original user formatting." -> Reference known issues: #397 and #490 
 * Polish the capitalization and punctuation

### PR DESCRIPTION
**Pull request by tjlaboss**
Fixes #498

### CODE REVIEW

**Comment by tjlaboss** [comment]
Feel free to push edits to this branch directly rather than doing a traditional review.

File: README.md [hunk 1]
```diff
@@ -25,21 +25,21 @@ see the [Installing section in the user guide](https://www.montepy.org/starting.
...
-* Parses Cells, surfaces, materials, and transforms very well.	
-* Can parse the following surfaces exactly P(X|Y|Z), C(X|Y|Z), C/(X|Y|Z) (I mean it can do PX, and PY, etc.)
-* Can read in all other inputs but not understand them	
-* Can write out full MCNP problem even if it doesn't fully understand an input.	
-* Can write out the MCNP problem verbatim, and try to match the original user formatting. 
+* Handles almost all MCNP input syntax.
+* Parses Cells, Surfaces, Materials, and Transforms very well.	
+* Can parse all surface types except macrobody facets ([Issue #354](https://github.com/idaholab/MontePy/issues/354)).
+* Can read and write out all other MCNP inputs, even if it doesn't not understand them	
+* Attempts to write out the MCNP problem verbatim, even matching the original user formatting. (See [Issues](https://github.com/idaholab/MontePy/issues).)
```
Comment (MicahGale) [inline README.md:39]: Issues is to broad I think.

File: README.md [hunk 2]
```diff
@@ -25,21 +25,21 @@ see the [Installing section in the user guide](https://www.montepy.org/starting.
 ## User Documentation
 
 MontePy has a [sphinx website](https://www.montepy.org/).
```
Comment (MicahGale) [inline README.md:27]: has a [documentation website for working with mcnp in python].

This is probably better for SEO.

File: README.md [hunk 3]
```diff
@@ -25,21 +25,21 @@ see the [Installing section in the user guide](https://www.montepy.org/starting.
...
-* Can write out the MCNP problem verbatim, and try to match the original user formatting. 
+* Handles almost all MCNP input syntax.
+* Parses Cells, Surfaces, Materials, and Transforms very well.	
+* Can parse all surface types except macrobody facets ([Issue #354](https://github.com/idaholab/MontePy/issues/354)).
+* Can read and write out all other MCNP inputs, even if it doesn't not understand them	
+* Attempts to write out the MCNP problem verbatim, even matching the original user formatting. (See [Issues](https://github.com/idaholab/MontePy/issues).)
 * Can quickly access cells, surfaces, and materials by their numbers. For example: `cell = problem.cells[105]`.
-* Can quickly update cell importances. For example `cell.importance.neutron = 2.0`.
-* Has over 240 test cases right now 
+* Can quickly update cell parameters, such as importances. For example `cell.importance.neutron = 2.0`.
```
Comment (MicahGale) [inline README.md:41]: add: 
* Can create universes, and fill cells with universes easily.

File: README.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ for cell in foo.cells:
   cell.number = i
```
Comment (MicahGale) [inline README.md:52]: replace

File: README.md [hunk 5]
```diff
@@ -62,16 +62,16 @@ Here a few of the known bugs and limitations:
 
 	
 * Cannot handle vertical input mode.
-* Does not support tallies in an easy way.
-* Does not support source definition in an easy way.
+* Does not support editing tallies in an easy way.
```
Comment (MicahGale) [inline README.md:65]: I don't like saying "easy"

**Review by MicahGale — COMMENTED** [review]
I'll make most of these edits myself, I just wanted to write down all my thoughts.

**Comment by MicahGale** [comment]
I have verified all links work. Could you review my changes @tjlaboss? Also I left one conversation open for you.

**Comment by MicahGale** [comment]
Do you think this closes #331?

File: README.md (hunk 1 above)
Comment (tjlaboss) [reply README.md:39]: I'm thinking leave the link, but change the text to say "See open issues"

**Comment by tjlaboss** [comment]
Let's leave #331 open for a larger effort

**Comment by tjlaboss** [comment]
Changes look good to me

**Review by MicahGale — APPROVED** [review]
My edits LGTM

**Pull request merged**